src/modules/list.js

```diff
@@ -0,0 +1,118 @@
+/* eslint-disable  no-use-before-define */
+
+let tasks = [];
+
+tasks.sort((a, b) => a.index - b.index);
+
+function saveTasksToLocalStorage() {
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+}
+
+function updateTaskIndexes() {
+  tasks.forEach((task, index) => {
+    task.index = index;
+  });
+}
+
+export function createItem() {
+  const list = document.getElementById('list');
+  list.innerHTML = '';
+  tasks.forEach((task) => {
+    const aTask = document.createElement('li');
+    list.appendChild(aTask);
+    aTask.classList.add('list-li');
+
+    aTask.addEventListener('click', () => {
+      makeTaskEditable(aTask, task);
+    });
+
+    aTask.textContent = task.description;
+    if (task.completed) {
+      aTask.classList.add('completed');
+    }
+  });
+}
+
+function deleteTask(task) {
+  const taskIndex = tasks.findIndex((item) => item === task);
+  if (taskIndex !== -1) {
+    tasks.splice(taskIndex, 1);
+    updateTaskIndexes();
+    saveTasksToLocalStorage();
+    createItem();
+  }
+}
+
+function makeTaskEditable(taskItem, task) {
+  const inputField = document.createElement('input');
+  inputField.type = 'text';
+  inputField.value = task.description;
+  inputField.classList.add('todo-input');
+  taskItem.innerHTML = '';
+  taskItem.appendChild(inputField);
+
+  const trashIcon = document.createElement('span');
+  trashIcon.classList.add('trash-icon');
+  trashIcon.innerHTML = '<i class="fa-solid fa-trash"></i>';
+
+  taskItem.appendChild(trashIcon);
+
+  inputField.addEventListener('keydown', (event) => {
+    if (event.key === 'Enter') {
+      task.description = inputField.value;
+      saveTasksToLocalStorage();
+      createItem();
+    }
+  });
+
+  trashIcon.addEventListener('click', (event) => {
+    event.stopPropagation();
+    deleteTask(task);
+  });
+
+  inputField.focus();
+}
+
+export function addTask(description) {
+  const newTask = {
+    description,
+    completed: false,
+    index: tasks.length,
+  };
```

Comment:
- [ ] Currently the index for each todo begins at `0`, rather it should begin at `1`. Following the requirement shown in the image below, if you have one task left in your array the index of it should be `1` but with your current implementation the index would be `0`. I kindly suggest you fix this by starting the index with 1, this would mean changing your code in line 80 to be the example below
```javascript
 index: tasks.length + 1,
```
![er](https://github.com/mohamedgamaleldin999999/An-interactive-to-do-list/assets/89055502/1812f372-8905-4fe3-816c-22a6d2af215c)